cmd/agent/app/flags_test.go

```diff
@@ -21,26 +21,31 @@
 package app
 
 import (
-	"flag"
 	"testing"
 
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestBingFlags(t *testing.T) {
 	cfg := NewBuilder()
-	flags := flag.NewFlagSet("test", flag.ContinueOnError)
-	cfg.Bind(flags)
-	err := flags.Parse([]string{
-		"-collector.host-port=1.2.3.4:555,1.2.3.4:666",
-		"-discovery.min-peers=42",
-		"-http-server.host-port=:8080",
-		"-processor.jaeger-binary.server-host-port=:1111",
-		"-processor.jaeger-binary.server-max-packet-size=4242",
-		"-processor.jaeger-binary.server-queue-size=42",
-		"-processor.jaeger-binary.workers=42",
+	command := cobra.Command{}
+	command.PersistentFlags().AddGoFlagSet(cfg.CreateFlags())
+	viper.BindPFlags(command.PersistentFlags())
+
+	err := command.ParseFlags([]string{
```

Comment:
Can you add a test for the env vars? Bonus points if you can add one for using a config file as well.